README.md

```diff
@@ -625,6 +626,18 @@ relevant file to trigger the incremental compiler. (Or run javac.)
 Alternatives
 ------------
 
+### Immutables vs `@FreeBuilder`
+
+[Immutables](https://immutables.github.io/) provides many of the same features as `@FreeBuilder`, plus a whole host more. Some are optional ways to potentially enhance performance, like [derived](https://immutables.github.io/immutable.html#derived-attributes) and [lazy](https://immutables.github.io/immutable.html#lazy-attributes) attributes, [singleton](https://immutables.github.io/immutable.html#singleton-instances) and [interned](https://immutables.github.io/immutable.html#instance-interning) instances, [hash code precomputation](https://immutables.github.io/immutable.html#precomputed-hashcode). Some are API-changing: [strict builders](https://immutables.github.io/immutable.html#strict-builder) provide a compile-time guarantee that all fields are set (but limit the use of builders as a consequence), while [copy methods](https://immutables.github.io/immutable.html#copy-methods) provide a concise way to clone a value with a single field changed (but require you to reference the generated ImmutableFoo type, not Foo). It provides advanced [binary serialization](https://immutables.github.io/immutable.html#serialization) options. It does not require getters follow the JavaBean naming convention (i.e. starting with `get` or `is`), and lets you easily customize the conventional method prefixes.
+
+There are some key differences between Immutables and `@FreeBuilder`:
+
+ * Immutables is *forward-only* in design: once items are added to collections, they cannot be removed. (In strict mode, even regular fields cannot be modified once they are set). This limits the use of the modify-rebuild pattern, which is increasingly useful as interfaces grow in complexity. You can work around this by clearing collections with the generated copy methods prior to creating a new builder.
```

Comment:
I think there are set methods for collections. So, you can modify them, but you have to modify the whole shebang.